packages/flame/lib/src/components/router_component.dart

```diff
@@ -104,18 +104,68 @@ class RouterComponent extends Component {
     _adjustRoutesVisibility();
   }
 
+  /// Pops the current route and places the route with [name] on top of the
+  /// navigation stack.
+  ///
+  /// If the route is already in the stack, it will simply be moved to the top.
+  /// Otherwise the route will be mounted and added at the top. The route's
+  /// page will also start building if it hasn't been built before. If the
+  /// route is already on top of the stack, this method will do nothing.
+  ///
+  /// This method calls the [Route.didPush] callback for the newly activated
+  /// route and also calls the [Route.didPop] callback for the popped route.
+  void pushReplacementNamed(String name) {
```

Comment:
Maybe it would be simpler to add a boolean flag to the `pushNamed()` method above?
```dart
void pushNamed(String name, {bool replaceCurrent = false}) {
  ...
}
```
As far as I can see, the difference between the two methods is just a single line, so it would make further maintenance easier to have just a single method instead of two.

Reply:
It's pretty good to use the same names as exist in Flutter though:
https://api.flutter.dev/flutter/widgets/Navigator/pushReplacementNamed.html

Reply:
I was going to do the flag originally,  but was like that definitely would be breaking, so avoided it, but then saw it in flutter so went that route. Unfortunately, I reversed word order but will fix that.

Reply:
It wouldn't be breaking since the flag is optional, but I think it is better to match Flutter's method names anyways.

Reply:
Ok, makes sense